### PR TITLE
Allow configuration of repository factory (as per ORM module)

### PR DESCRIPTION
**Pull request by Bombdog**
The need has arisen for me to inject a caching service into my repositories. I've found some examples for the ORM module but I noticed that the equivalent is not available for the ODM module.

http://blog.tomhanderson.com/2016/01/dependency-injection-in-doctrine.html

I checked and the RepositoryFactory was added in to ODM on 28th March 2015. It would be nice to be able to use this.

https://github.com/doctrine/mongodb-odm/pull/892

### CODE REVIEW

**Comment by malarzm** [comment]
FWIW the feature is available in all stable versions

**Comment by Bombdog** [comment]
Sorry I should have used a branch rather than master. Do I need to do anything? This is rather old :-(

**Comment by alcaeus** [comment]
@Bombdog you don't need to do anything, I'll cherry-pick the commit manually should need be.

**Comment by alcaeus** [comment]
@Bombdog can you add tests for the new config option? Thanks!

**Comment by Bombdog** [comment]
@alcaeus I added an additional assertion to the existing configuration factory test. Followed the style of what was there already.

File: src/DoctrineMongoODMModule/Options/Configuration.php [hunk 1]
```diff
@@ -420,4 +426,20 @@ public function getTypes()
     {
         return $this->types;
     }
+
+    /**
+     * @return string
+     */
+    public function getRepositoryFactory() {
```
Comment (alcaeus) [inline src/DoctrineMongoODMModule/Options/Configuration.php]: Apparently we don't have automatic checks here, but the opening curly brace goes in its own line for functions (see PSR-2).

File: src/DoctrineMongoODMModule/Options/Configuration.php [hunk 2]
```diff
@@ -420,4 +426,20 @@ public function getTypes()
...
+        return $this->repositoryFactory;
+    }
+
+    /**
+     * @param string $repositoryFactory
+     * @return \DoctrineMongoODMModule\Options\Configuration
+     */
+    public function setRepositoryFactory($repositoryFactory) {
```
Comment (alcaeus) [inline src/DoctrineMongoODMModule/Options/Configuration.php]: Curly brace again.

File: src/DoctrineMongoODMModule/Service/ConfigurationFactory.php [hunk 3]
```diff
@@ -85,6 +85,11 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
             $config->setClassMetadataFactoryName($factoryName);
         }
 
+        // respositoryFactory, if set
+        if ($repositoryFactory = $options->getRepositoryFactory()){
```
Comment (alcaeus) [inline src/DoctrineMongoODMModule/Service/ConfigurationFactory.php]: Missing a space before the curly brace

**Review by alcaeus — CHANGES_REQUESTED** [review]

**Comment by Bombdog** [comment]
Ok I just amended, cheers

**Review by alcaeus — APPROVED** [review]

**Comment by alcaeus** [comment]
Thanks, @Bombdog!

**Pull request merged**